js/src/apbct-public--5--external-forms.js

```diff
@@ -292,9 +292,81 @@ window.onload = function() {
         ctProtectExternal();
         catchDynamicRenderedForm();
         catchNextendSocialLoginForm();
+        ctProtectOutsideIframe();
     }, 2000);
 };
 
+/**
+ * Protect forms placed in iframe with outside src
+ */
+function ctProtectOutsideIframe() {
+    let iframes = document.querySelectorAll('iframe');
+    if (iframes.length > 0) {
+        iframes.forEach(function(iframe) {
+            if (iframe.src.indexOf('form.typeform.com') !== -1 ||
+                iframe.src.indexOf('forms.zohopublic.com') !== -1 ||
+                iframe.src.indexOf('link.surepathconnect.com') !== -1
+            ) {
+                ctProtectOutsideIframeHandler(iframe);
+            }
+        });
+    }
+}
+
+let ctProtectOutsideIframeCheck;
+/**
+ * Protect forms placed in iframe with outside src handler
+ * @param {HTMLElement} iframe
+ */
+function ctProtectOutsideIframeHandler(iframe) {
+    let cover = document.createElement('div');
+    cover.style.width = '100%';
+    cover.style.height = '100%';
+    cover.style.background = 'black';
+    cover.style.opacity = 0;
+    cover.style.position = 'absolute';
+    cover.style.top = 0;
+    cover.onclick = function(e) {
+        if (ctProtectOutsideIframeCheck === undefined) {
+            let currentDiv = e.currentTarget;
+            currentDiv.style.opacity = 0.5;
+            let preloader = document.createElement('div');
+            preloader.className = 'apbct-iframe-preloader';
+            currentDiv.appendChild(preloader);
+
+            let data = {
```

Comment:
What about event_token? We should check if event_token is available on the page and add it to the data if so. 
```
            const token = apbctLocalStorage.get('bot_detector_event_token');
```

Also, we must generate event_token somehow for this page if no forms available  besides the iframe

Reply:
The bot detector carries additional information only if there are additional forms on the page. Therefore, it makes no sense to generate it if there are no other forms.

Reply:
So, if another form exists added event token to request body https://github.com/CleanTalk/wordpress-antispam/pull/335/commits/1fc2dd0c12276785b65150edb4846f6ca94f6935